packages/core/src/indexing-store/realtime.ts

```diff
@@ -163,7 +164,10 @@ export const createRealtimeIndexingStore = ({
                               .map(({ js }) => table[js]),
                             set: valuesU,
                           })
-                          .returning();
+                          .returning()
+                          .then((res) =>
+                            Array.isArray(values) ? res : res[0],
```

Comment:
So, `res` can either be a single row or an array (matching `values`)?

Seems like we don't have type safety here, so would be good to make sure each of these branches works as expected. Do you think it's worthwhile to update the tests to also validate return objects?